modules/drivers/athena/src/metabase/driver/athena.clj

```diff
@@ -315,58 +315,50 @@
     (catch Exception e
       (log/error (u/format-color 'red "Failed to execute query: %s %s" query (.getMessage e))))))
 
-(defn- describe-database->clj
-  "Workaround for wrong getColumnCount response by the driver (huh?)"
-  [column-metadata]
-  {:name (str/trim (:col_name column-metadata))
-   :type (str/trim (:data_type column-metadata))})
-
-(defn- remove-invalid-columns
-  "Returns a transducer."
-  []
-  (comp (remove #(= (:col_name %) ""))
-        (remove #(= (:col_name %) nil))
-        (remove #(= (:data_type %) nil))
-        (remove #(str/starts-with? (:col_name %) "#")) ; remove comment
-        (distinct)                                     ; driver can return twice the partitioning fields
-        (map describe-database->clj)))
+(defn- valid-column? [column-name database-type]
+  (and (seq column-name)
+       (seq database-type)
+       ;; remove comment
+       (not (str/starts-with? column-name "#"))))
+
+(defn- table-columns [driver ^DatabaseMetaData metadata ^String db-name-or-nil ^String schema ^String table-name]
+  (with-open [rset (.getColumns metadata db-name-or-nil schema table-name nil)]
+    (loop [acc #{}]
```

Comment:
if we're doing all this, is it worth our time to use transients?

Reply:
I reworked this to separate out the fetching from resultset with the reducing code, not it's using `(into #{} ...)` which uses transients under the hood